pkg/generic/generic.go

```diff
@@ -76,14 +83,22 @@ func MapThriftGenericForJSON(p DescriptorProvider) (Generic, error) {
 //
 //	g, err := generic.HTTPThriftGeneric(p)
 //	SetBinaryWithBase64(g, true)
-func HTTPThriftGeneric(p DescriptorProvider) (Generic, error) {
-	codec, err := newHTTPThriftCodec(p, thriftCodec)
-	if err != nil {
-		return nil, err
+func HTTPThriftGeneric(p DescriptorProvider, opts ...DynamicgoOptions) (Generic, error) {
```

Comment:
DynamicgoOptions seems not universal, refactor it by referring to Option defined in internel/client/option.go.